samples/client/petstore/R/R/api_client.R

```diff
@@ -87,8 +93,43 @@ ApiClient  <- R6::R6Class(
       if (!is.null(timeout)) {
         self$timeout <- timeout
       }
+
+      if (!is.null(retryStatusCodes)) {
+        self$retryStatusCodes <- retryStatusCodes
+      }
+	  
+      if (!is.null(maxRetryAttempts)) {
+        self$maxRetryAttempts <- maxRetryAttempts
+      }
     },
+
     CallApi = function(url, method, queryParams, headerParams, body, ...){
+
+      resp <- self$Execute(url, method, queryParams, headerParams, body, ...)
+      statusCode <- httr::status_code(resp)
+
+      if(is.null(self$maxRetryAttempts)){
+        self$maxRetryAttempts = 3
+      }
+      
+	  if(!is.null(self$retryStatusCodes)){
+
+        for(i in 1: self$maxRetryAttempts){
+          if(statusCode %in% self$retryStatusCodes){
+            Sys.sleep((2 ^ i) + stats::runif(n = 1, min = 0, max = 1))
+            resp <- self$Execute(url, method, queryParams, headerParams, body, ...)
+            statusCode <- httr::status_code(resp)
+          }
+          else{
+            break;
+          }
+        }	  
+	  }
```

Comment:
The line indention doesn't look right to me. It seems to be using a mix of tabs and spaces